BUG : when a user does multiple votes the test is expected to reverted with message "User Voted" but the contract reverts with a custom error

while running the tests one test fails  with :  ElectionFactory and Election Contracts
                                                                           Voting Process
                                                                                Should allow voting and ensure only one vote per user:
                                                                      AssertionError: Expected transaction to be reverted with reason 'User Voted', but it reverted with a custom 
                                                                       error
                                                                       at Context.<anonymous> (test\ElectionTest.js:177:7)

this happens because the test  expects to be reverted with "User Voted" but the contract reverts with a custom error 

![Image](https://github.com/user-attachments/assets/b28ff0e5-ed9a-4b91-aff7-f1852d91a739)

![Image](https://github.com/user-attachments/assets/ed57b65e-e492-43fa-aa40-72c8241084d2)

--- a/blockchain/test/ElectionTest.js
+++ b/blockchain/test/ElectionTest.js
@@ -176,7 +176,7 @@ describe("ElectionFactory and Election Contracts", function () {
 
       await expect(
         electionInstance.connect(voter1).userVote([0])
-      ).to.be.revertedWith("User Voted");
+    ).to.be.revertedWithCustomError(electionInstance, "AlreadyVoted");
     });
   });
 

